api.proto

```diff
@@ -48,7 +48,16 @@ message FetchMetadataResponse {
 
 // PublishRequest is sent to publish a new message.
 message PublishRequest {
-    Message message = 1; // Message to publish
+    bytes               key           = 1;  // Message key
+    bytes               value         = 2;  // Message payload
+    string              stream        = 3;  // Stream name to publish to
```

Comment:
I assume this just doesn't get set if using the low-level publish direct to a subject?

Reply:
If it is set, it's ignored since the subject is more specific